src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioDiagnosticAnalyzerProvider.WorkspaceEventListener.cs

```diff
@@ -84,9 +80,17 @@ private async Task<VisualStudioDiagnosticAnalyzerProvider> CreateProviderAsync()
                 // this will allow us to build once and deploy on different versions of VS SxS.
                 var vsDteVersion = Version.Parse(dte.Version.Split(' ')[0]); // DTE.Version is in the format of D[D[.D[D]]][ (?+)], so we need to split out the version part and check for uninitialized Major/Minor below
 
-                var assembly = Assembly.Load($"Microsoft.VisualStudio.ExtensionManager, Version={(vsDteVersion.Major == -1 ? 0 : vsDteVersion.Major)}.{(vsDteVersion.Minor == -1 ? 0 : vsDteVersion.Minor)}.0.0, PublicKeyToken=b03f5f7f11d50a3a");
-                var typeIExtensionContent = assembly.GetType("Microsoft.VisualStudio.ExtensionManager.IExtensionContent");
-                var type = assembly.GetType("Microsoft.VisualStudio.ExtensionManager.SVsExtensionManager");
+                // TODO: Remove all the reflection based code in this method and VisualStudioDiagnosticAnalyzerProvider.
+                var assemblyVersion = $"{(vsDteVersion.Major == -1 ? 0 : vsDteVersion.Major)}.{(vsDteVersion.Minor == -1 ? 0 : vsDteVersion.Minor)}.0.0";
+                var extensionManagerAssembly = Assembly.Load($"Microsoft.VisualStudio.ExtensionManager, Version={assemblyVersion}, PublicKeyToken=b03f5f7f11d50a3a");
+                var extensionEngineAssembly = Assembly.Load($"Microsoft.VisualStudio.ExtensionEngine, Version={assemblyVersion}, PublicKeyToken=b03f5f7f11d50a3a");
```

Comment:
Hmm, so digging in a bit more, it seems this assembly is itself a new one, I don't see it on a prior VS2022 intpreview build. I guess I need to guard here against assembly load failing `Microsoft.VisualStudio.ExtensionEngine`. I am going to try this change locally on an older VS2022 build without the ExtensionEngine assembly.

Reply:
Ah, my bad. `Microsoft.VisualStudio.ExtensionEngine` is indeed present on older builds as well, it was just in a different folder earlier (`Common7\IDE` instead of `Common7\IDE\PrivateAssemblies`). We should be good here.